docs/adr/004_region_as923_implementation.md

```diff
@@ -2,34 +2,118 @@
-- DwellTime - it is not yet clear whether the parameter needs to be set or not, this is tracked in user story [#732](https://github.com/Azure/iotedge-lorawan-starterkit/issues/732)
 - FOpts - it is not yet clear whether this parameter needs to be used, this is tracked for all regions as part of user story [#717](https://github.com/Azure/iotedge-lorawan-starterkit/issues/717)
 
 ## Decision
 
+### Frequency offsets
+
 The `AS923_FREQ_OFFSET` parameter can be calculated based on the channel 0 and channel 1 frequencies in the LoRa Basics Station configuration. The corresponding channels for region AS923 are defined as follows:
 
 `Channel 0 frequency Hz = 923200000 + AS923_FREQ_OFFSET_HZ`
 `Channel 1 frequency Hz = 923400000 + AS923_FREQ_OFFSET_HZ`
 
 Using this formula we will calculate the offset by subtracting 923200000 from the configured channel 0 frequency. We will use the formula for channel 1 frequency to validate the offset value and throw and exception if values are not the same.
 
-In the implementation of region AS923 the frequencies for channel 0 and 1 will be passed to the region-specific constructor where the offset value will be calculated.  
+In the implementation of region AS923 the frequencies for channel 0 and 1 will be passed to the region-specific constructor where the offset value will be calculated.
+
+### Dwell Times
+
+We will support dwell times through the automatic dwell time management process described in the appendix. We will not support the automated cache refresh in a first version, but we will document the limitation that for multi-gateway support the device cache either needs to be manually invalidated or updated by waiting until it expires.
+
+## Appendix
+
+### Dwell times
+
+We discuss different approaches to handle dwell time limitations.
+
+#### No `TxParamSetupReq` support
+
+If we do not support the `TxParamSetupReq` MAC Command as part of the initial version of the AS923 region implementation. One implication is that the maximum Equivalent Isotropically Radiated Power (EIRP), which is normally configurable through `TxParamSetupReq`, will be static and non-configurable. We introduce a Boolean configuration value in the concentrator device twin that specifies whether a dwell time limitation applies to that concentrator. Based on this configuration value, the gateway will apply a different set of regional parameters, which take the dwell time limitations into account. It will be the responsibility of the starter kit user to ensure that end devices are configured with the same dwell time settings as the concentrator to which they are connected, as it will not be automatically propagated to the end device through `TxParamSetupReq`.
+
+Pros:
+
+- Easy to implement
+
+Cons:
+
+- End devices (e.g. by Netvox) cannot be updated with different dwell time limitation defaults after shipping. If local regulations change, such devices cannot be used any longer with the starter kit
```

Comment:
That's a valid argument but I don't quite get how it's only applicable to this case and not the other options? Can the dwell time setting on these devices still be modified with a MAC command?

Reply:
Yes, that can be done using `TxParamSetupReq`.